core/src/main/java/org/elasticsearch/gateway/DanglingIndicesState.java

```diff
@@ -108,37 +124,77 @@ void cleanupAllocatedDangledIndices(MetaData metaData) {
      * to the currently tracked dangling indices.
      */
     void findNewAndAddDanglingIndices(MetaData metaData) {
```

Comment:
we should probably rename this method to indicates it searches for dangling and deleted indices.